### PR TITLE
[CSBS] Refactor & API Update

### LINKED ISSUE #459: [CSBS] API Updated
https://confluence.tsi-dev.otc-service.com/display/DR/RM-4122-CSBS-API

### PR DESCRIPTION
**Pull request by Aloento**
Move all API methods and related structures to a separate file.
Inline all `Extracts` and use the `extract` package instead.
Remove all Builders

fix #459

### CODE REVIEW

**Comment by Aloento** [comment]
@anton-sidelnikov @artem-lifshits please review this thank you

File: openstack/csbs/v1/backup/results.go [hunk 1]
```diff
@@ -1,90 +1,114 @@
...
-}
-
-type ResourceCapability struct {
-	Result       bool   `json:"result"`
-	ResourceType string `json:"resource_type"`
-	ErrorCode    string `json:"error_code"`
-	ErrorMsg     string `json:"error_msg"`
-	ResourceId   string `json:"resource_id"`
-}
+const ProviderID = "fc4d5750-22e7-4798-8a46-f48f62c4c1da"
```
Comment (anton-sidelnikov) [inline openstack/csbs/v1/backup/results.go:7]: i think you can remove that

File: openstack/csbs/v1/policies/requests.go [hunk 2]
```diff
@@ -1,227 +1,120 @@
 package policies
```
Comment (anton-sidelnikov) [inline openstack/csbs/v1/policies/requests.go]: this filename misleading, maybe combine it with results in one file?

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: openstack/csbs/v1/backup/results.go (hunk 1 above)
Comment (Aloento) [reply openstack/csbs/v1/backup/results.go:7]: no, still using

**Comment by Aloento** [comment]
recheck

**Comment by Aloento** [comment]
recheck

**Comment by Aloento** [comment]
recheck

File: openstack/csbs/v1/backup/results.go (hunk 1 above)
Comment (anton-sidelnikov) [reply openstack/csbs/v1/backup/results.go:7]: omg, this is hardcoded value?

Comment (Aloento) [reply openstack/csbs/v1/backup/results.go:7]: it is

File: openstack/csbs/v1/policies/requests.go (hunk 2 above)
Comment (Aloento) [reply openstack/csbs/v1/policies/requests.go]: done

**Review by anton-sidelnikov — APPROVED** [review]

**Comment by Aloento** [comment]
@artem-lifshits @Polina-Gubina

**Comment by Aloento** [comment]
recheck

**Comment by Aloento** [comment]
@artem-lifshits @Polina-Gubina please check thanks

**Review by artem-lifshits — APPROVED** [review]

**Pull request merged**